tests/unittests/GraphGradTest.cpp

```diff
@@ -208,3 +208,32 @@ TEST(GraphAutoGrad, checkConvertToGradTest) {
   EE.compile(CompilationMode::Train, TF);
   EE.compile(CompilationMode::Infer, F);
 }
+
+/// Check that we can differentiate functions that use MatMulNode.
+TEST(GraphAutoGrad, checkMatMulToGradTest) {
```

Comment:
How does this test show that the gradient for MatMul is correct? The graph gets differentiated and compiled, but it is never run.